oauth2client/contrib/_metadata.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Provides helper methods for talking to the Compute Engine metadata server.
+
+See https://cloud.google.com/compute/docs/metadata
+"""
+
+import datetime
+import httplib2
+import json
+
+from six.moves import http_client
+from six.moves.urllib.parse import urlencode
+
+from oauth2client._helpers import _from_bytes
+from oauth2client.client import _UTCNOW
+
+METADATA_ROOT = 'http://metadata.google.internal/computeMetadata/v1/'
+METADATA_HEADERS = {'Metadata-Flavor': 'Google'}
+
+
+def get(path, http_request=None, root=METADATA_ROOT, **kwargs):
```

Comment:
I still think that's super ugly. Can I keep it this way if we make `get` a module internal function?